tests/unit/CMakeLists.txt

```diff
@@ -0,0 +1,565 @@
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+set(COMMON_SOURCES
+    test_platform.cpp
+    test_util.cpp
+)
+
+set(COMMON_LIBS
+    openthread-ftd
+    openthread-ncp-ftd
+    mbedtls
+    util
+)
+
+set(COMMON_INCLUDES
+    ${PROJECT_SOURCE_DIR}/include
+    ${PROJECT_SOURCE_DIR}/src
+    ${PROJECT_SOURCE_DIR}/src/core
+    ${PROJECT_SOURCE_DIR}/examples/platforms/simulation
+)
+
+set(COMMON_COMPILE_OPTIONS
+    -DOPENTHREAD_FTD=1
+    -DOPENTHREAD_SPINEL_CONFIG_OPENTHREAD_MESSAGE_ENABLE=1
+)
+
+add_executable(test-aes
+    ${COMMON_SOURCES}
+    test_aes.cpp
+)
+
+add_executable(test-child
+    ${COMMON_SOURCES}
+    test_child.cpp
+)
+
+add_executable(test-child-table
+    ${COMMON_SOURCES}
+    test_child_table.cpp
+)
+
+add_executable(test-flash
+    ${COMMON_SOURCES}
+    test_flash.cpp
+)
+
+add_executable(test-heap
+    ${COMMON_SOURCES}
+    test_heap.cpp
+)
+
+add_executable(test-hmac-sha256
+    ${COMMON_SOURCES}
+    test_hmac_sha256.cpp
+)
+
+add_executable(test-ip6-address
+    ${COMMON_SOURCES}
+    test_ip6_address.cpp
+)
+
+add_executable(test-link-quality
+    ${COMMON_SOURCES}
+    test_link_quality.cpp
+)
+
+add_executable(test-linked-list
+    ${COMMON_SOURCES}
+    test_linked_list.cpp
+)
+
+add_executable(test-lowpan
+    ${COMMON_SOURCES}
+    test_lowpan.cpp
+)
+
+add_executable(test-mac-frame
+    ${COMMON_SOURCES}
+    test_mac_frame.cpp
+)
+
+add_executable(test-message
+    ${COMMON_SOURCES}
+    test_message.cpp
+)
+
+add_executable(test-message-queue
+    ${COMMON_SOURCES}
+    test_message_queue.cpp
+)
+
+add_executable(test-netif
+    ${COMMON_SOURCES}
+    test_netif.cpp
+)
+
+add_executable(test-network-data
+    ${COMMON_SOURCES}
+    test_network_data.cpp
+)
+
+add_executable(test-priority-queue
+    ${COMMON_SOURCES}
+    test_priority_queue.cpp
+)
+
+add_executable(test-pskc
+    ${COMMON_SOURCES}
+    test_pskc.cpp
+)
+
+add_executable(test-string
+    ${COMMON_SOURCES}
+    test_string.cpp
+)
+
+add_executable(test-timer
+    ${COMMON_SOURCES}
+    test_timer.cpp
+)
+
+target_include_directories(test-aes
```

Comment:
Should we group these by executable name rather than cmake command? That would make it easier to see what affects a given executable.